pkg/azure/aks.go

```diff
@@ -57,6 +63,89 @@ func ListClusters(ctx context.Context, subscriptionId, resourceGroup string) ([]
 	return clusters, nil
 }
 
+func LinkAcr(ctx context.Context, subscriptionId, clusterResourceGroup, clusterName, acrResourceGroup, acrName string) error {
+	lgr := logger.FromContext(ctx).With("subscription", subscriptionId, "resource group", clusterResourceGroup, "cluster name", clusterName,
+		"acr name", acrName)
+	ctx = logger.WithContext(ctx, lgr)
+	lgr.Debug("linking ACR")
+
+	// add validation for acr?
+
+	client, err := aksFactory(subscriptionId)
+	if err != nil {
+		return fmt.Errorf("getting aks client: %w", err)
+	}
+
+	lgr.Debug("getting cluster information")
+	cluster, err := client.NewManagedClustersClient().Get(ctx, clusterResourceGroup, clusterName, nil)
+
+	if err != nil {
+		return fmt.Errorf("getting cluster information: %w", err)
+	}
+
+	var assigneeId *string
+
+	if cluster.Identity == nil {
+		return fmt.Errorf("serviceprincipal clusters are not supported at this time")
+		//lgr.Debug("detected service principal cluster")
```

Comment:
should we remove the commented code?

Reply:
want to leave it for if/when we need to support sp clusters